contest/contest_info.py

```diff
@@ -97,10 +99,13 @@ def get_scoreboard(contest):
 
     return scoreboard
 
-def get_clarifications(contest,user):
-    reply_all = Clarification.objects.filter(contest = contest, reply_all = True)
-    user_ask = Clarification.objects.filter(contest = contest, asker = user)
-    return reply_all | user_ask
+def get_clarifications(user,contest):
+    if has_contest_ownership(user,contest):
+        return Clarification.objects.filter(contest = contest)
+    else:
+        reply_all = Clarification.objects.filter(contest = contest, reply_all = True)
+        user_ask = Clarification.objects.filter(contest = contest, asker = user)
+        return reply_all | user_ask
 
 def is_contestant(user,contest):
```

Comment:
Adding that annotation can reduce the chance we query DB and redirdect to login qage